vars/buildExecute.groovy

```diff
@@ -102,7 +108,12 @@ void call(Map parameters = [:]) {
                     error "[${STEP_NAME}] buildTool not set and no dockerImage & dockerCommand provided."
                 }
         }
-
+        if (config.cnbBuild) {
```

Comment:
What if `buildTool` is `docker` and `cnbBuild=true`? line 97 would still call `kanikoExecute` and this would throw an exception, right?

Reply:
> Yes- is that what we want to happen?

No. Not at all. I think we want the same to happen as in `piper`, don't we? So why "invent" something new?

In short:

* If `buildTool: docker`, we check `cnbBuild: true/false` to check if the image should be build with `kaniko` or `cnbBuild`.
* For the other `buildTool`s, we create an image after the build if `cnbBuild: true`.
* *BUT* We should check also for the supported `buildTool`s (currently `npm`, `maven`, `gradle` and `mta`).

WDYT?

Reply:
That sounds good! What do you think of this: https://github.com/SAP/jenkins-library/pull/4498/files#r1305461395

Reply:
I've implemented it. What do you think?